File: Java/Petshop.java
class PetShop {
    private int id;
    private String namaProduk;
    private int hargaProduk;
    private int stokProduk;

    public PetShop() {
        this.id = 0;
        this.namaProduk = "";
        this.hargaProduk = 0;
        this.stokProduk = 0;
    }

    public PetShop(int id, String namaProduk, int hargaProduk, int stokProduk) {
        this.id = id;
        this.namaProduk = namaProduk;
        this.hargaProduk = hargaProduk;
        this.stokProduk = stokProduk;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public void setNamaProduk(String namaProduk) {
        this.namaProduk = namaProduk;
    }

    public String getNamaProduk() {
        return namaProduk;
    }

    public void setHargaProduk(int hargaProduk) {
        this.hargaProduk = hargaProduk;
    }

    public int getHargaProduk() {
        return hargaProduk;
    }

    public void setStokProduk(int stokProduk) {
        this.stokProduk = stokProduk;
    }

    public int getStokProduk() {
        return stokProduk;
    }

    public void tampilkanData(int panjangId, int panjangNama, int panjangHarga, int panjangStok) {
        System.out.printf("%-" + panjangId + "d%-" + panjangNama + "s%-" + panjangHarga + "d%-" + panjangStok + "d", id, namaProduk, hargaProduk, stokProduk);
    }
}

File: CPP/Petshop.cpp
#include <iostream>
#include <iomanip>
#include <string>
#include <list>
#include <vector>

using namespace std;

class PetShop {
private:
    int id;
    string nama_produk;
    int harga_produk;
    int stok_produk;

public:
    PetShop() {
        this->id = 0;
        this->nama_produk = "";
        this->harga_produk = 0;
        this->stok_produk = 0;
    }

    PetShop(int id, string nama_produk, int harga_produk, int stok_produk) {
        this->id = id;
        this->nama_produk = nama_produk;
        this->harga_produk = harga_produk;
        this->stok_produk = stok_produk;
    }

    void setId(int id) {this->id = id;}
    int getId() const {return id;}

    void setNamaProduk(string nama_produk) {this->nama_produk = nama_produk;}
    string getNamaProduk() const {return nama_produk;}

    void setHargaProduk(int harga_produk) {this->harga_produk = harga_produk;}
    int getHargaProduk() const {return harga_produk;}

    void setStokProduk(int stok_produk) {this->stok_produk = stok_produk;}
    int getStokProduk() const {return stok_produk;}

    void tampilkanData(int panjang_id, int panjang_nama, int panjang_harga, int panjang_stok) const {
        cout << left << setw(panjang_id) << id
             << setw(panjang_nama) << nama_produk
             << setw(panjang_harga) << harga_produk
             << setw(panjang_stok) << stok_produk;
    }

    ~PetShop() {}
};